src/tests/payment/test_stripe_webhook.py

```diff
@@ -0,0 +1,371 @@
+                                    "active": True,  # Plan status
+                                },
+                            },
+                        ],
+                    },
+                }
+            },
+        }
+
+        # Create a CustomerDetail object to simulate the Stripe user
+        CustomerDetail.objects.create(
+            user=user, stripe_customer_id=payload["data"]["object"]["customer"]
+        )
+
+        # Create an initial subscription (before the update)
+        Subscription.objects.create(
+            user=user,
+            cancel_at_period_end=False,
+            current_period_start=timezone.now(),
+            current_period_end=timezone.now() + timedelta(days=30),
+            plan=True,
+        )
+
+        # Generate the signature header for Stripe (adapt according to your logic)
+        sig_header = generate_stripe_signature(payload, STRIPE_WEBHOOK_SECRET)
+        mock_construct_event.return_value = payload  # Simulate the Stripe event
+        # API webhook call
+        response = client.post(
+            reverse_lazy("v1:payment:stripe-webhook"),  # Ensure the URL is correct
+            data=json.dumps(payload),
+            content_type="application/json",
+            HTTP_STRIPE_SIGNATURE=sig_header,  # Header signature
+        )
+
+        # Verify the response is correct
+        assert response.status_code == 200
+
+        customer = CustomerDetail.objects.get(
+            stripe_customer_id=payload["data"]["object"]["customer"]
+        )
+        user = customer.user
+
+        # Retrieve the subscription after the update
+        subscription = Subscription.objects.get(user=user)
+
+        # Check that the subscription values are updated as in the Stripe event
+        assert (
+            subscription.cancel_at_period_end
+            == payload["data"]["object"]["cancel_at_period_end"]
+        )
+        assert subscription.trial_end == (
+            None
+            if payload["data"]["object"]["trial_end"] == "null"
+            else timezone.make_aware(
+                datetime.fromtimestamp(payload["data"]["object"]["trial_end"])
+            )
+        )
+        assert subscription.plan == bool(
+            payload["data"]["object"]["items"]["data"][0]["plan"]["active"]
+        )
+
+    @mock.patch("stripe.Webhook.construct_event")
+    def test_invoice_paid(self, mock_construct_event, client, user):
+        """
+        Test the `invoice.paid` event.
+        Verify that the `hosted_invoice_url` and `invoice_pdf` fields
+        of the subscription are correctly updated.
+        """
+        payload = {
+            "type": "invoice.paid",
+            "data": {
+                "object": {
+                    "customer": "cus_REbNQXKKFCRF2c",
+                    "hosted_invoice_url": "https://stripe.com/invoice/test123",
+                    "invoice_pdf": "https://stripe.com/invoice/test123.pdf",
+                }
+            },
+        }
```

Comment:
same here, maybe we want to extract that payload out of the function so it's grouped with the other. It might as well be a fixture if we wanted

Reply:
I've extracted all the payloads from the functions and grouped them at the beginning of the file